lib/workload/stateless/stacks/sample-sheet-check/sample-sheet-check-lambda/src/samplesheet.py

```diff
@@ -108,6 +108,14 @@ def check_sample_name_format(self):
             logger.error("Sample ID {} did not have corresponding Sample_Name".format(self.sample_id))
             raise SampleNameFormatError
 
+        # Check if the Sample_Name column is the library_id from the Sample_ID column ignoring the "_topup" / "_rerun"
+        clean_unique_id = SAMPLE_REGEX_OBJS["topup"].sub('', self.unique_id)
```

Comment:
I think there's a better way to regex this, will have a closer look in the morning.

Reply:
Actually because I used the non-capturing group syntax in almost all of the regexes `(?:..)` this is probably the most appropriate way forward